internal/pkg/service/common/etcdop/op/txn.go

```diff
@@ -298,132 +323,166 @@ func (v *lowLevelTxn[R]) addPart(ctx context.Context, part txnPart[R]) error {
-				r.AddSubResult(NoResult{})
-				return
-			}
-
-			// Call original mapper of the sub transaction
-			if subResult, err := lowLevel.MapResponse(ctx, r.Response().SubResponse(subTxnResponse.OpResponse())); err == nil {
-				r.AddSubResult(subResult)
-			} else {
-				r.AddSubResult(err).AddErr(err)
-			}
-		})
 	default:
 		panic(errors.Errorf(`unexpected operation type "%s"`, part.Type))
 	}
 
 	return nil
 }
 
-func (v *lowLevelTxn[R]) addThen(op etcd.Op, mapper MapFn) int {
+func (v *lowLevelTxn[R]) addThen(op etcd.Op, mapper MapFn) {
 	v.thenOps = append(v.thenOps, op)
 
 	// If the transaction is successful (then branch), the OP result will be available in the Responses[INDEX],
 	// so we can call original mapper with the sub-response.
-	index := len(v.thenOps) - 1
 	if mapper != nil {
+		index := len(v.thenOps) - 1
 		v.processors = append(v.processors, func(ctx context.Context, r *TxnResult[R]) {
 			if r.Succeeded() {
 				rawSubResponse := r.Response().Txn().Responses[index]
-				subResponse := r.Response().SubResponse(mapRawResponse(rawSubResponse))
-				if subResult, err := mapper(ctx, subResponse); err == nil {
-					r.AddSubResult(subResult)
-				} else {
+				subResponse := r.Response().WithOpResponse(mapRawResponse(rawSubResponse))
+				if _, err := mapper(ctx, subResponse); err != nil {
 					r.AddErr(err)
 				}
 			}
 		})
 	}
-
-	return index
 }
 
-func (v *lowLevelTxn[R]) addElse(op etcd.Op, mapper MapFn) int {
+func (v *lowLevelTxn[R]) addElse(op etcd.Op, mapper MapFn) {
 	v.elseOps = append(v.elseOps, op)
 
 	// If the transaction is NOT successful (else branch), the OP result will be available in the Responses[INDEX],
 	// so we can call original mapper with the sub-response.
-	index := len(v.elseOps) - 1
 	if mapper != nil {
+		index := len(v.elseOps) - 1
 		v.processors = append(v.processors, func(ctx context.Context, r *TxnResult[R]) {
 			if !r.Succeeded() {
-				rawSubResponse := r.Response().Txn().Responses[index]
-				subResponse := r.Response().SubResponse(mapRawResponse(rawSubResponse))
-				if subResult, err := mapper(ctx, subResponse); err == nil {
-					r.AddSubResult(subResult)
-				} else {
+				subResponse := r.Response().WithOpResponse(mapRawResponse(r.Response().OpResponse().Txn().Responses[index]))
+				if _, err := mapper(ctx, subResponse); err != nil {
 					r.AddErr(err)
 				}
 			}
 		})
 	}
-
-	return index
 }
 
-func (v *lowLevelTxn[R]) mapResponse(ctx context.Context, raw RawResponse) *TxnResult[R] {
-	// Map transaction response
-	r := newTxnResult(&raw, v.result)
-	r.succeeded = raw.Txn().Succeeded
+func (v *lowLevelTxn[R]) mergeTxn(ctx context.Context, op Op) error {
+	// Step down to a nested Merge operation, if there is no processor/else branch.
+	if txn, ok := op.(txnInterface); ok && isSimpleTxn(txn) {
+		thenStart := len(v.thenOps)
+		elseStart := len(v.elseOps)
+		if err := v.addParts(ctx, txn); err != nil {
+			return err
+		}
+
+		if txn.txnProcessorsCount() > 0 {
```

Comment:
What is counted as txn processor count. Is it just `if else` branch or also other commands?